internal/auth/auth.go

```diff
@@ -62,19 +65,42 @@ func NewAuthService(logger logr.Logger, cluster *kubernetes.Cluster) *AuthServic
 	}
 }
 
-// GetUsers returns all the Epinio users
-func (s *AuthService) GetUsers(ctx context.Context) ([]User, error) {
+// GetUsers returns all the Epinio users with no conflicting definitions. it further returns a map
+// of definition counts enabling the caller to distinguish between `truly does not exist` versus
+// `has conflicting definitions`.
+func (s *AuthService) GetUsers(ctx context.Context) ([]User, DefinitionCount, error) {
 	s.Logger.V(1).Info("GetUsers")
 
 	secrets, err := s.getUsersSecrets(ctx)
 	if err != nil {
-		return nil, errors.Wrap(err, "error getting users secrets")
+		return nil, nil, errors.Wrap(err, "error getting users secrets")
 	}
 
 	users := []User{}
 	usernames := []string{}
 
+	// map of per-user definition counts, to detect users with conflicting definitions
+	userCount := DefinitionCount{}
+
+	for _, secret := range secrets {
+		name := string(secret.Data["username"])
+		count, ok := userCount[name]
+		if !ok {
+			userCount[name] = 1
+			continue
+		}
+		userCount[name] = count + 1
```

Comment:
The zero value of a int is already 0, so this can be changed to:

```suggestion
		count := userCount[name]
		userCount[name] = count + 1
```

Reply:
Actually, then we do not need a helper variable either. ok

Reply:
done. commit b04ac0d8192ab565f309eb40f53dc5acf34710cb